docs/spec/typeddict.rst

```diff
@@ -227,195 +480,181 @@ Additional notes on TypedDict class inheritance:
-  ``'x'`` could be present and have an :term:`inconsistent <consistent>` type
-  (which may not be visible through ``A`` due to :term:`structural`
-  assignability). Example::
+    extra_int: MovieExtraInt = {"name": "Blade Runner", "year": 1982}
+    int_mapping: Mapping[str, int] = extra_int  # Not OK. 'int | str' is not assignable with 'int'
+    int_str_mapping: Mapping[str, int | str] = extra_int  # OK
 
-      class A(TypedDict, total=False):
-          x: int
-          y: int
+As a consequence, every TypedDict type is :term:`assignable` to ``Mapping[str, object]``.
 
-      class B(TypedDict, total=False):
-          x: int
+.. _typeddict-dict:
 
-      class C(TypedDict, total=False):
-          x: int
-          y: str
+Subtyping with ``dict``
+^^^^^^^^^^^^^^^^^^^^^^^
 
-      def f(a: A) -> None:
-          a['y'] = 1
+Generally, TypedDict types are not subtypes of any specialization of ``dict[...]`` type, since
+dictionary types allow destructive operations, including ``clear()``. They
+also allow arbitrary keys to be set, which would compromise type safety.
 
-      def g(b: B) -> None:
-          f(b)  # Type check error: 'B' not assignable to 'A'
+However, a TypedDict with :term:`extra items` may be a subtype of ``dict[str, VT]``,
+provided certain conditions are met, because it introduces sufficient restrictions
+for this subtyping relation to be safe.
+A TypedDict type is a subtype of ``dict[str, VT]`` if the following conditions are met:
 
-      c: C = {'x': 0, 'y': 'foo'}
-      g(c)
-      c['y'] + 'bar'  # Runtime error: int + str
+- The TypedDict type has mutable :term:`extra items` of a type that is :term:`equivalent` to ``VT``.
+- All items on the TypedDict satisfy the following conditions:
+  - The value type of the item is :term:`equivalent` to ``VT``.
+  - The item is not read-only.
+  - The item is not required.
 
-* A TypedDict isn't :term:`assignable` to any ``Dict[...]`` type, since
-  dictionary types allow destructive operations, including ``clear()``.  They
-  also allow arbitrary keys to be set, which would compromise type safety.
-  Example::
+For example::
 
-      class A(TypedDict):
-          x: int
+    class IntDict(TypedDict, extra_items=int):
+        pass
 
-      class B(A):
-          y: str
+    class IntDictWithNum(IntDict):
+        num: NotRequired[int]
 
-      def f(d: Dict[str, int]) -> None:
-          d['y'] = 0
+    def f(x: IntDict) -> None:
+        v: dict[str, int] = x  # OK
+        v.clear()  # OK
 
-      def g(a: A) -> None:
-          f(a)  # Type check error: 'A' not assignable to Dict[str, int]
+    not_required_num_dict: IntDictWithNum = {"num": 1, "bar": 2}
+    regular_dict: dict[str, int] = not_required_num_dict  # OK
+    f(not_required_num_dict)  # OK
 
-      b: B = {'x': 0, 'y': 'foo'}
-      g(b)
-      b['y'] + 'bar'  # Runtime error: int + str
+In this case, methods that are previously unavailable on a TypedDict are allowed,
```

Comment:
nit: "previously" made sense when it meant "prior to this PEP", but here these methods do not become available on the TypedDict. In the context of the spec, we can replace "previously" with "generally".